### PR TITLE
Update Cluster Loader design

### PR DESCRIPTION
**Pull request by wojtek-t**

### CODE REVIEW

File: clusterloader/docs/design.md [hunk 1]
```diff
@@ -48,7 +48,7 @@ A single test scenario will be defined by a `Config`. Its schema will be as foll
 struct Config {
 	// TBD: Some config allowing to access cluster, e.g. path to kubeconfig file.
 
-	// Number of namespace in the test.
+	// Number of (default) namespaces in the test.
```
Comment (shyamjvs) [inline clusterloader/docs/design.md]: I don't understand. Isn't 'default' namespace only one?

Comment (wojtek-t) [reply clusterloader/docs/design.md]: I guess it's overloaded term.
I didn't mean the k8s 'default' namespace.

I meant a "default set of namespace" that you will get out of the box already created, deleted at the end and managed for you.

Do you suggest different name?

Comment (shyamjvs) [reply clusterloader/docs/design.md]: I read below. The use of 'default' is a bit confusing here. Maybe say something like 'No. of namespaces you want the test to create'?

File: clusterloader/docs/design.md [hunk 2]
```diff
@@ -185,21 +185,27 @@ In many cases, it will be enough for users for many namespaces to contain the sa
...
+- In the top-level test definition, we will define the number of default namespaces
+  that will be used by a test.
+- Default namespaces will have names of a form “namespace-<number>” for number
   in range 1..N (where N is the number of namespaces in a test)
 
-To make it usable, we introduce the following type:
+However, users may want to create their own namespaces (as part of `Phases`) and
+create objects in them. That is perfectly valid usecase that will be supported.
+
+To make it possible to reference a set of namespace (both default and user-created),
```
Comment (shyamjvs) [inline clusterloader/docs/design.md]: namespaces

File: clusterloader/docs/design.md [hunk 3]
```diff
@@ -302,22 +308,32 @@ struct ResourceGatherer {
...
 }
-funct (r *ResourceGatherer) StopMeasurement() {
+func (r *ResourceGatherer) StopMeasurement() {
 	// Stop gatherer.
 	// Validate and/or report results.
 }
 ```
 
-and the registering it in some config file, will let you automatically use:
+and the registering this type in some factory, to enable use of:
```
Comment (shyamjvs) [inline clusterloader/docs/design.md]: delete 'the'

File: clusterloader/docs/design.md (hunk 1 above)
Comment (wojtek-t) [reply clusterloader/docs/design.md]: Changed to sth different - PTAL.

File: clusterloader/docs/design.md (hunk 2 above)
Comment (wojtek-t) [reply clusterloader/docs/design.md]: done.

File: clusterloader/docs/design.md (hunk 3 above)
Comment (wojtek-t) [reply clusterloader/docs/design.md]: done

File: clusterloader/docs/design.md [hunk 4]
```diff
@@ -185,21 +185,27 @@ In many cases, it will be enough for users for many namespaces to contain the sa
 objects (or to be more specific: objects created from the same templates). Obviously,
 we want the config for the test to be as small as possible.
 As a result, we will introduce the following rules:
-- In the top-level test definition, we will define the number of namespaces that will
-  be used by a test.
-- All the namespaces will have names of a form “namespace-<number>” for number
+- In the top-level test definition, we will define the number of default namespaces
```
Comment (shyamjvs) [inline clusterloader/docs/design.md]: 'number of nmespaces that will be automanaged by clusterloader'?

File: clusterloader/docs/design.md [hunk 5]
```diff
@@ -185,21 +185,27 @@ In many cases, it will be enough for users for many namespaces to contain the sa
...
+  that will be used by a test.
+- Default namespaces will have names of a form “namespace-<number>” for number
```
Comment (shyamjvs) [inline clusterloader/docs/design.md]: Default -> The automanaged

File: clusterloader/docs/design.md [hunk 6]
```diff
@@ -185,21 +185,27 @@ In many cases, it will be enough for users for many namespaces to contain the sa
...
 ```
 struct NamespaceRange {
 	Min int32
 	Max int32
+	Basename *string
 }
 ```
 
-Such object defines all namespaces "namespace-\<i\>" for `i` in the range [Min, Max].
+Unset `Basename` will be defaulted to 'namespace'. Such object defines all namespaces
```
Comment (shyamjvs) [inline clusterloader/docs/design.md]: Could you change this line to:

```
The `NamespaceRange` would select all namespaces "\<Basename\>-\<i\>" for `i` in the range [Min, Max]. If `Basename` is unset, it'll be defaulted to the basename used for automanaged namespaces (i.e `namespace`).
```

File: clusterloader/docs/design.md (hunk 4 above)
Comment (wojtek-t) [reply clusterloader/docs/design.md]: done

File: clusterloader/docs/design.md (hunk 5 above)
Comment (wojtek-t) [reply clusterloader/docs/design.md]: done

File: clusterloader/docs/design.md (hunk 6 above)
Comment (wojtek-t) [reply clusterloader/docs/design.md]: done

File: clusterloader/docs/design.md [hunk 7]
```diff
@@ -186,20 +186,27 @@ objects (or to be more specific: objects created from the same templates). Obvio
 we want the config for the test to be as small as possible.
 As a result, we will introduce the following rules:
 - In the top-level test definition, we will define the number of namespaces that will
-  be used by a test.
-- All the namespaces will have names of a form “namespace-<number>” for number
+  be automanaged by ClusterLoader.
+- The automanaged will have names of a form “namespace-<number>” for number
```
Comment (shyamjvs) [inline clusterloader/docs/design.md]: -> 'automanaged namespaces will have names of the'

**Comment by shyamjvs** [comment]
One nit and lgtm.

**Pull request merged**